docs/sources/clients/promtail/scraping.md

```diff
@@ -360,6 +360,44 @@ scrape_configs:
 Only `api_token` and `zone_id` are required.
 Refer to the [Cloudfare](../../configuration/#cloudflare) configuration section for details.
 
+## Object store scraping
+
+Promtail supports scraping logs from Object store. Currently promtail supports scraping logs from S3.
+
+### How scraping object store works with AWS S3?
+
+S3 should be integrated with SQS and Upload and Put events should be sent to SQS queue. Promtail polls SQS for the messages and on receiving the messages it parses the message to get the `objects` which were uploaded. 
+Then we pull each object, read the log lines and send the lines to Loki server. Messages will be deleted/acknowledged only after the object is completely being read.  
+
+Things to note on how objectstore treats objects in few scenarios
+
+1. It's possible that we will receive the same object again from the queue for which the reader is still active. In this case we check the modified date time of the object. If it's same as the last known time we skip creating one more reader for it. If the modified date time is changed, we stop the active reader and create new reader. Also, the line or position to start reading depends on `reset_cursor` configuration. If set to `true` objects will be read from begining otherwise it will be read from last saved cursor position. 
+2. If promtail is stopped or crashed, the objects which were not completely read will resume the reading from the last saved position of the cursor. Since, messages are deleted/acknowledged only after the the object is completely read, on restarting the promtail we receive the same messages again from the queue and we start the reading from the last saved position.
+
+
+### Next steps
+
+In the next versions of object scraping, we would like to support other Object storage providers like GCS, Swift etc...
+
```

Comment:
Our documentation should not be suggesting future changes.  That goes into issues and design docs.

```suggestion
```